DP02_09_Custom_Coadds/DP02_09b_Custom_Coadd_Sources.ipynb

```diff
@@ -345,371 +336,329 @@
    "id": "93271ee8-d9d0-4a1d-81d8-e5f9f42635b7",
    "metadata": {},
    "source": [
-    "# 2. Run source detection and measurement"
+    "# 2. Run source detection, deblending, and measurement"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "565a9b2b-eda4-4c48-8d31-6ca9d3b33bdf",
+   "id": "974799d4-ea3d-4639-a7cb-470fad3500f9",
    "metadata": {},
    "source": [
-    "## 2.1. Create schema"
+    "This tutorial uses an approach to running source detection that is meant to parallel the approach adopted in tutorial notebook 09a for coaddition: use the <a href=\"https://pipelines.lsst.io/py-api/lsst.ctrl.mpexec.SimplePipelineExecutor.html\">Simple Pipeline Executor</a> to run a subset of Tasks from the DP0.2 Data Release Production (DRP) pipeline. The usage of SimplePiplineExecutor here also closely parallels the command line version of source detection demonstrated in DP0.2 <a href=\"https://dp0-2.lsst.io/tutorials-examples/cmdline-custom-coadd.html\">command line tutorial 02</a>."
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "1c122eaa-2c38-4008-981e-2f6db4b27b72",
-   "metadata": {},
-   "source": [
-    "Create a basic schema to use with the source measurement and detection tasks."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "06a83779-a435-41f9-ad18-57c542ca6abb",
+   "id": "16f3600d-17c0-4790-bc87-56490109a7e7",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "schema = afwTable.SourceTable.makeMinimalSchema()\n",
-    "print(schema)"
+    "## 2.1. Set up an output collection for the source detection results"
```

Comment:
Can the output collection defined in 2.1 be the same as the input collection in this case? As in, could we add sources to our existing collection instead of storing them in a new collection?

Reply:
Unfortunately, no. I tried running with the output collection being the same as the input collection, and it failed at

```
simpleButler = SimplePipelineExecutor.prep_butler(butler_repo, 
                                                  inputs=[inputCollection], 
                                                  output=my_outputCollection)
```

With an error about `Cycle in collection chaining`. I think this has to do with the same forum post that I linked to in the command line version of this tutorial regarding the need for a different output collection versus input collection.

> ---------------------------------------------------------------------------
> ValueError                                Traceback (most recent call last)
> Input In [14], in <cell line: 1>()
> ----> 1 simpleButler = SimplePipelineExecutor.prep_butler(butler_repo, 
>       2                                                   inputs=[inputCollection], 
>       3                                                   output=my_outputCollection)
> 
> File /opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-4.1.0/Linux64/ctrl_mpexec/g993939a6a1+f01622d10f/python/lsst/ctrl/mpexec/simple_pipeline_executor.py:110, in SimplePipelineExecutor.prep_butler(cls, root, inputs, output, output_run)
>     108 collections = [output_run]
>     109 collections.extend(inputs)
> --> 110 butler.registry.setCollectionChain(output, collections)
>     111 # Remake butler to let it infer default data IDs from collections, now
>     112 # that those collections exist.
>     113 return Butler(butler=butler, collections=[output], run=output_run)
> 
> File /opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-4.1.0/Linux64/daf_butler/g35a551b10e+17c7ad0751/python/lsst/daf/butler/core/utils.py:56, in transactional.<locals>.inner(self, *args, **kwargs)
>      53 @functools.wraps(func)
>      54 def inner(self: Any, *args: Any, **kwargs: Any) -> Any:
>      55     with self.transaction():
> ---> 56         return func(self, *args, **kwargs)
> 
> File /opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-4.1.0/Linux64/daf_butler/g35a551b10e+17c7ad0751/python/lsst/daf/butler/registries/sql.py:387, in SqlRegistry.setCollectionChain(self, parent, children, flatten)
>     385 children = CollectionSearch.fromExpression(children)
>     386 if children != record.children or flatten:
> --> 387     record.update(self._managers.collections, children, flatten=flatten)
> 
> File /opt/lsst/software/stack/stack/miniconda3-py38_4.9.2-4.1.0/Linux64/daf_butler/g35a551b10e+17c7ad0751/python/lsst/daf/butler/registry/interfaces/_collections.py:204, in ChainedCollectionRecord.update(self, manager, children, flatten)
>     200 for record in children.iter(
>     201     manager, flattenChains=True, includeChains=True, collectionTypes={CollectionType.CHAINED}
>     202 ):
>     203     if record == self:
> --> 204         raise ValueError(f"Cycle in collection chaining when defining '{self.name}'.")
>     205 if flatten:
>     206     children = CollectionSearch.fromExpression(
>     207         tuple(record.name for record in children.iter(manager, flattenChains=True))
>     208     )
> 
> ValueError: Cycle in collection chaining when defining 'u/ameisner/custom_coadd_window1_test1'.
>

Reply:
I added a reference to the relevant community forum post in commit [9d50051](https://github.com/rubin-dp0/tutorial-notebooks/pull/161/commits/9d500517b7b42cd99f68ebdfebfaee3b69442749). Will proceed to mark this comment as resolved.